### ISSUE TITLE
Share operator does not reconnect to source observable when rxjs spy is running

### ISSUE THREAD

**Issue by jpmitche11** [cartant/rxjs-spy#37]
Rxjs-spy is not handling connectable observables correctly.  This will impact the multicast family of operators, including share.

When a shared observable is completely unsubscribed, then resubscribed later, normally the source observable is resubscribed.
When there is a spy object created, this is not working.  The share operator creates an internal subject, I see that is not being unsubscribed as well.

Here is some code that demonstrates the issue.  The issue is present if a spy object has been created.

```
const shared$ = interval(100).pipe(
  share()
);

const a = shared.subscribe(() => log('a'))
setTimeout(() => a.unsubscribe(), 200);  // a logged twice, then the interval() observable is unsubscribed

setTimeout(() => shared.subscribe(() => log('b')),  1000);    //Nothing logged. This should log b repeatedly

```

Here is a runnable version
https://stackblitz.com/edit/rxjs-spy-share?file=index.ts

**Comment by cartant** [comment]
Thanks. I really appreciate the StackBlitz reproduction.

**Comment by cartant** [comment]
Until I find and fix the problem, there is an alternate workaround. You can hide the shared observable from the spy, like this:

```ts
import { hide } from "rxjs-spy/operators";
const shared$ = interval(100).pipe(
  share(),
  hide()
);
```

**Comment by jpmitche11** [comment]
Confirmed hide operator works around the issue.  Thanks.

**Comment by cartant** [comment]
This should be fixed in 7.0.3.
